cpp/log/etcd_consistent_store-inl.h

```diff
@@ -238,55 +240,38 @@ bool LessBySequenceNumber(const EntryHandle<Logged>& lhs,
 
 
 template <class Logged>
-util::Status EtcdConsistentStore<Logged>::GetSequencedEntries(
-    std::vector<EntryHandle<Logged>>* entries) const {
-  util::Status status(GetAllEntriesInDir(GetFullPath(kSequencedDir), entries));
-  sort(entries->begin(), entries->end(), LessBySequenceNumber<Logged>);
-  if (status.ok()) {
-    for (const auto& entry : *entries) {
-      CHECK(entry.Entry().has_sequence_number());
-    }
+void EtcdConsistentStore<Logged>::CheckMappingIsOrdered(
+    const ct::SequenceMapping& mapping) const {
+  if (mapping.mapping_size() < 2) {
+    return;
+  }
+  int64_t seq(mapping.mapping(0).sequence_number());
+  for (int64_t i = 1; i < mapping.mapping_size(); ++i) {
+    const int64_t entry_seq(mapping.mapping(i).sequence_number());
+    CHECK_EQ(entry_seq, seq + 1);
+    seq = entry_seq;
   }
-  return status;
 }
 
 
 template <class Logged>
-util::Status EtcdConsistentStore<Logged>::GetSequencedEntry(
-    const int64_t sequence_number, EntryHandle<Logged>* entry) const {
-  CHECK_GE(sequence_number, 0);
-  util::Status status(GetEntry(GetSequencedPath(sequence_number), entry));
-  if (status.ok()) {
-    CHECK(entry->Entry().has_sequence_number());
+util::Status EtcdConsistentStore<Logged>::GetSequenceMapping(
+    EntryHandle<ct::SequenceMapping>* sequencing) const {
+  util::Status status(GetEntry(GetFullPath(kSequenceFile), sequencing));
+  if (!status.ok()) {
+    return status;
   }
-
-  return status;
+  CheckMappingIsOrdered(sequencing->Entry());
```

Comment:
Maybe we should also check that the sequence number of the first entry is less than or equal to the serving STH tree size?